tests/ui/wrong_self_convention.rs

```diff
@@ -88,3 +88,29 @@ mod issue4037 {
         }
     }
 }
+
+// Lint also in trait definition (see #6307)
+mod issue6307 {
+    trait T: Sized {
```

Comment:
Let's modify some tests by not providing the function body to check that methods that are only declared are also linted.

Reply:
I think we should apply the conventions for this lint whatever the supertrait is, so your suggestion to add test when no body is given is right IMO.